src/main/java/com/intel/gkl/compression/IntelDeflaterFactory.java

```diff
@@ -31,14 +31,17 @@ public IntelDeflaterFactory() {
      */
     public Deflater makeDeflater(final int compressionLevel, final boolean gzipCompatible) {
         if (intelDeflaterSupported) {
-            if ((compressionLevel == 1 && gzipCompatible) || compressionLevel != 1) {
-                return new IntelDeflater(compressionLevel, gzipCompatible);
-            }
-        }
-        logger.warn("IntelDeflater is not supported, using Java.util.zip.Deflater");
-        return new Deflater(compressionLevel, gzipCompatible);
-    }
+	      try {
+                    return new IntelDeflater(compressionLevel, gzipCompatible); 
+              } catch  (IllegalArgumentException e) {
+                    logger.warn("Invalid configuration requsted, using Java.util.zip.Deflater");
+	      }
```

Comment:
Maybe we should use a different exception class here? If we pass to IntelDeflater/Deflater constructor incorrect values (compressionLevel > 9 or compressionLevel < -1) it will raise IllegalArgumentException and It will be also caught in this try-catch statement.

Reply:
In this case, the execution will go to JavaDeflater and the IllegalArgumentException will be reraised but not caught this time. Is it ok?

Reply:
Java will catch and return it's errors in this case.